### PR TITLE
Building a Tiny JS World

### PR DESCRIPTION
**Pull request by maximmorenko**
### Building a Tiny JS World

[Demo](https://maximmorenko.github.io/building-a-tiny-js-world-master) |
[Code base](https://github.com/maximmorenko/building-a-tiny-js-world-master)

### CODE REVIEW

File: submissions/morenko/building-a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    species: 'cat',
+    name: 'Tom',
+    gender: 'male',
+    legs: 4,
+    hands: 0,
+    saying: 'mau',
+  };
+  
+  
+  const inhabitants = [dog, cat, woman, man,];
```
Comment (artem-trubin) [inline submissions/morenko/building-a-tiny-js-world/index.js:35]: There is no reason to leave a trailing coma if a whole array is on the same line.

File: submissions/morenko/building-a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const woman = {
+    species: 'human',
+    name: 'line',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    saying: 'bye',
+  };
```
Comment (artem-trubin) [inline submissions/morenko/building-a-tiny-js-world/index.js:8]: I'm going to write this once, but you have to fix indentations in the whole file.

File: submissions/morenko/building-a-tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  
+  const inhabitants = [dog, cat, woman, man,];
+  
+  const keys = ['species', 'name', 'gender', 'legs', 'hands', 'saying'];
+  
+  const description = inhabitants.map((el) => {
+    return keys.map(key => el[key]);
+  });
+  
+  description.map(el => print(el.join('; ')));
```
Comment (artem-trubin) [inline submissions/morenko/building-a-tiny-js-world/index.js:43]: Add an empty line in the end of the file.

File: submissions/morenko/building-a-tiny-js-world/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    hands: 0,
+    saying: 'mau',
+  };
+  
+  
+  const inhabitants = [dog, cat, woman, man,];
+  
+  const keys = ['species', 'name', 'gender', 'legs', 'hands', 'saying'];
+  
+  const description = inhabitants.map((el) => {
```
Comment (artem-trubin) [inline submissions/morenko/building-a-tiny-js-world/index.js:39]: Maybe there could be a better name for an inhabitant than `el`. `el` and `key` are a little too generic.

**Review by artem-trubin — COMMENTED** [review]
Hi! Nice work, let's make it a little better.

**Comment by artem-trubin** [comment]
Indentations are still wrong. You should close `}` on the same level where you open it. And trailing comas in arrays that are placed on several lines is good, why did you remove them?

**Comment by maximmorenko** [comment]
я видимо неправильно понял предыдущее замечание по запятым) спасибо! исправил. А если объект записать в одну строчку, то запятая после последней пары ключ-значение тоже будет лишняя? или это касается только массивов? как здесь например:
const man = {hands: 2, saying: 'bye'};
const man = {
hands: 2, 
saying: 'bye',
};
оба объекта верно записаны?

**Comment by artem-trubin** [comment]
Yes, both objects are writter right. This trailing coma is useful when you need to add something in this _list_ of things. If you don't leave the coma, add something (with a coma of course) and check it in a git diff, you'll see both lines effected, because you added a coma to the existing one. It adds a little confusion. Though, sometimes you should not leave a trailing coma, like, for example, JSON syntax required you to remove trailing comas. You did it right now.

**Comment by OleksiyRudenko** [comment]
@maximmorenko we need to fix the following situation:
image

Please submit a PR that will:
1. remove the code with incompliant path naming
2. add the same code with compliant path naming
3. clearly says it is a correction PR

**Pull request merged**